core/drivers/crypto/caam/hal/common/hal_cfg.c

```diff
@@ -0,0 +1,293 @@
+#include "hal_jr.h"
+
+#ifndef CFG_CAAM_DT
+/* Hal includes */
+#include "hal_ctrl.h"
+#endif // CFG_CAAM_DT
+
+#ifdef CFG_CAAM_DT
+static const char *dt_ctrl_match_table = {
+	"fsl,sec-v4.0-ctrl",
+};
+
+static const char *dt_jr_match_table = {
+	"fsl,sec-v4.0-job-ring",
+};
+
+/**
+ * @brief   Get the DT interrupt property of the node.
+ *
+ * @param[in] fdt   Reference to the Device Tree
+ * @param[in] node  Node offset
+ *
+ * @retval  Interrupt number if value >= 0
+ */
+static int dt_get_irq(void *fdt, int node)
+{
+	const uint32_t *int_prop;
+	int len_prop;
+	int it_num;
+
+	/*
+	 * Interrupt property is defined with 3x32 bits word:
+	 *  - Type of interrupt
+	 *  - Interrupt Number
+	 *  - Interrupt Trigger type
+	 */
+	int_prop = fdt_getprop(fdt, node, "interrupts", &len_prop);
+
+	if ((int_prop == NULL) || (len_prop < 2)) {
+		HAL_TRACE("Interrupts property error (0x%08x)", len_prop);
+		return (-1);
+	}
+
+	it_num = fdt32_to_cpu(int_prop[1]);
+	HAL_TRACE("Interrupts number %d", it_num);
+
+	return it_num;
+}
+
+/**
+ * @brief   Finds the Job Ring reserved for the Security
+ *
+ * @param[in]     fdt         Reference to the Device Tree
+ * @param[in]     status      Status flag of the node to found
+ * @param[in,out] start_node  Node offset to start in DTB, current node found
+ *
+ * @retval  JR offset reads in the DTB (reg property) if found.
+ * @retval  0 if Job Ring not reserved
+ */
+static paddr_t dt_find_jr_index(void *fdt, int status, int *start_node)
+{
+	paddr_t jr_offset = 0;
+	int node;
+	int node_status;
+
+	node = fdt_node_offset_by_compatible(fdt, *start_node,
+			dt_jr_match_table);
+
+	while (node != -FDT_ERR_NOTFOUND) {
+		node_status = _fdt_get_status(fdt, node);
+		HAL_TRACE("Found Job Ring node status 0x%x", node_status);
+		if ((node_status & status) == status) {
+			HAL_TRACE("Found Job Ring node @%d", node);
+			jr_offset   = _fdt_reg_base_address(fdt, node);
+			*start_node = node;
+			break;
+		}
+
+		node = fdt_node_offset_by_compatible(fdt, node,
+				dt_jr_match_table);
+	}
+
+	HAL_TRACE("JR Base address return 0x%"PRIxPTR"", jr_offset);
+	return jr_offset;
+}
+#endif // CFG_CAAM_DT
+
+/**
+ * @brief   Returns the Job Ring Configuration to be used by the TEE
+ *
+ * @param[out] jr_cfg   Job Ring Configuration
+ *
+ * @retval  CAAM_NO_ERROR   Success
+ * @retval  CAAM_FAILURE    An error occurred
+ */
+enum CAAM_Status hal_cfg_get_conf(struct jr_cfg *jr_cfg)
+{
+	enum CAAM_Status retstatus = CAAM_FAILURE;
+
+	vaddr_t ctrl_base;
+
+#ifdef CFG_CAAM_DT
+
+	paddr_t jr_offset;
+	size_t  size;
+
+	void *fdt;
+	int  node;
+
+	fdt = get_dt();
+	if (!fdt) {
+		EMSG("DTB no present\n");
+		goto exit_get_conf;
+	}
+
+	node = fdt_node_offset_by_compatible(fdt, 0, dt_ctrl_match_table);
+
+	if (node < 0) {
+		EMSG("Caam Node not found err = 0x%X\n", node);
+		goto exit_get_conf;
+	}
+
+#ifdef CFG_DT_MODIFY
+	/* Ensure that CAAM Control is secure-status enabled */
+	if (dt_set_secure_status(fdt, node)) {
```

Comment:
`dt_set_secure_status` does not exist in this patch set nor in the OP-TEE master.

Reply:
Correct. This is why it's built only if the CFG_DT_MODIFY is define. This will  require a specific patch to add this function. Do you agree to keep it for now while a new patch is submitterd?

Reply:
Either remove this function or add the implementation, this is up to you. However I think most of the dt modification functions contained in the HAL are general enough to be included in the OP-TEE core files instead of the CAAM hal.